### PR TITLE
Add support FOSRestBundle 2.0

### PR DESCRIPTION
**Pull request by DuchkinaNast**
I am targetting this branch, because i want to add support FOSRestBundle v2.0.

## Changelog

```markdown
### Added
 - added support for `FOSRestBundle:2.0`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 .php_cs.cache
 composer.lock
 phpunit.xml
+/.idea/
```
Comment (core23) [inline .gitignore]: Nope!

File: .idea/workspace.xml [hunk 2]
```diff
@@ -0,0 +1,466 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (core23) [inline .idea/workspace.xml]: No no no! Please don't IDE settings!

File: .gitignore (hunk 1 above)
Comment (greg0ire) [inline .gitignore]: Please remove this. Instead, use a global gitignore, and exclude this directory for all your git repo : https://github.com/greg0ire/dotfiles/blob/master/gitconfig#L33

File: Controller/Api/BlockController.php [hunk 3]
```diff
@@ -26,6 +24,8 @@
  */
 class BlockController
 {
+    use FOSRestApiTrait;
```
Comment (greg0ire) [inline Controller/Api/BlockController.php]: We support php 5.3, so we don't have access to traits, sorry… use inheritance instead.

**Comment by greg0ire** [comment]
If you want to start over, 
- `git reset origin/3.x`
- redo your commits
- `git push -f`

File: Controller/Api/FOSRestApiTrait.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+use JMS\Serializer\SerializationContext;
+
+trait FOSRestApiTrait
+{
+    /**
+     * @param ParamFetcherInterface $paramFetcher
+     *
+     * @return ParamFetcherInterface
+     */
+    public function setMapForOrderByParam(ParamFetcherInterface $paramFetcher)
```
Comment (core23) [inline Controller/Api/FOSRestApiTrait.php]: Why public? Can we make this protected and/or final?

File: .gitignore (hunk 1 above)
Comment (DuchkinaNast) [inline .gitignore]: Thanks for your help!!! Sorry for this mistake. It's my first pull request

File: Controller/Api/FOSRestController.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+use JMS\Serializer\SerializationContext;
+
+class FOSRestController
+{
+    /**
+     * @param ParamFetcherInterface $paramFetcher
+     *
+     * @return ParamFetcherInterface
+     */
+    protected function setMapForOrderByParam(ParamFetcherInterface $paramFetcher)
```
Comment (greg0ire) [inline Controller/Api/FOSRestController.php]: `final protected`

File: Controller/Api/FOSRestController.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        return $paramFetcher;
+    }
+
+    /**
+     * @param $entity
+     * @param array $groups
+     *
+     * @return FOSRestView
+     */
+    protected function serializeContext($entity, array $groups)
```
Comment (greg0ire) [inline Controller/Api/FOSRestController.php]: `final protected`

File: .gitignore (hunk 1 above)
Comment (greg0ire) [inline .gitignore]: No problem, thanks a lot for contributing this!

File: Controller/Api/FOSRestController.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+namespace Sonata\PageBundle\Controller\Api;
+
+use FOS\RestBundle\Context\Context;
+use FOS\RestBundle\Controller\Annotations\QueryParam;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\View\View as FOSRestView;
+use JMS\Serializer\SerializationContext;
+
+class FOSRestController
```
Comment (OskarStark) [inline Controller/Api/FOSRestController.php]: `@author` missing

File: Controller/Api/SiteController.php [hunk 8]
```diff
@@ -61,16 +60,17 @@ public function __construct(SiteManagerInterface $siteManager, FormFactoryInterf
...
-     * @View(serializerGroups="sonata_api_read", serializerEnableMaxDepthChecks=true)
+     * @View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)
      *
      * @param ParamFetcherInterface $paramFetcher
      *
      * @return PagerInterface
      */
     public function getSitesAction(ParamFetcherInterface $paramFetcher)
     {
+        $this->setMapForOrderByParam($paramFetcher);
```
Comment (OskarStark) [inline Controller/Api/SiteController.php]: I would add a blank line afterwards

Comment (DuchkinaNast) [inline Controller/Api/SiteController.php]: Thanks! I've done it

File: Controller/Api/FOSRestController.php [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+namespace Sonata\PageBundle\Controller\Api;
+
+use FOS\RestBundle\Context\Context;
+use FOS\RestBundle\Controller\Annotations\QueryParam;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\View\View as FOSRestView;
+use JMS\Serializer\SerializationContext;
+
+/**
+ * Class FOSRestController.
```
Comment (OskarStark) [inline Controller/Api/FOSRestController.php]: please remove this comment

**Comment by Soullivaneuh** [comment]
The changelog need to be reviewed.

The changed section is not here just to list changed files.

BTW, the changelog must more focus on what we changed/fixed instead of how we did it.

**Comment by greg0ire** [comment]
As seen on other bundles, you might need to mock the paramfetcher implementation rather than its interface, which does not have `addParam` IIRC

**Comment by DuchkinaNast** [comment]
@core23 @greg0ire @Soullivaneuh @OskarStark  Thanks for all of you! 
Could you review it once more? Is everything okay now?

File: Controller/Api/FOSRestController.php [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+use FOS\RestBundle\Context\Context;
+use FOS\RestBundle\Controller\Annotations\QueryParam;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\View\View as FOSRestView;
+use JMS\Serializer\SerializationContext;
+
+/**
+ * @author Duchkina Anastasiya <duchkina.nast@gmail.com>
+ */
+class FOSRestController
```
Comment (core23) [inline Controller/Api/FOSRestController.php]: Maybe make this abstract?

**Comment by OskarStark** [comment]
ping @DuchkinaNast

**Comment by greg0ire** [comment]
The CHANGELOG should be way, way more concise. You should be talking to users, not developers of the library, so a single line saying support for FOSRest 2 was added should be enough.

**Comment by greg0ire** [comment]
ping @DuchkinaNast

**Comment by DuchkinaNast** [comment]
@greg0ire , @OskarStark Sorry guys, was soooo busy!!! @greg0ire Thanks for your advice! I hope that everything is correct now.

**Comment by greg0ire** [comment]
@core23 please review again, thanks @DuchkinaNast

**Comment by greg0ire** [comment]
ping @core23

**Comment by greg0ire** [comment]
Thanks @DuchkinaNast !

**Pull request merged**